bika/lims/browser/worksheet/views/analyses.py

```diff
@@ -439,3 +447,42 @@ def get_slot_header(self, item):
 
         pos_text += "</table>"
         return pos_text
+
+    def fill_remarks_cells(self, items):
+        """Gets the Remarks from the parent AR for each slot.
+        :param items: dictionary with items to be rendered in the list
+        """
+        filled_slots = []
+        for item in items:
+            item_position = item['Pos']
+            if item_position in filled_slots:
+                # We've already filled the remarks cell for this slot
+                continue
+            if item['state_title'] == 's':
+                # This is an empty slot
+                continue
+            filled_slots.append(item_position)
+
+            # This is the first analysis found for the given position, add the
+            # parent's remarks and apply rowspan.
+            rowspan = self.items_rowspans.get(item_position, 1)
+            item['rowspan']['Remarks'] = rowspan
+            item['replace']['Remarks'] = self.get_slot_remarks(item)
+
+            # Set the css class to it's default hidden state
+            item['class']['Remarks'] = 'hidden'
+
+
+    def get_slot_remarks(self, item):
+        """Return the remarks for the analysis' parent.
+        No remarks are returned for Duplicate analyses.
+        :param item: the item for which the slot header is requested
+        :return: the html contents to be displayed in the remarks cell of a slot
+        """
+        instance = api.get_object(item['obj'])
+        parent = instance.aq_parent
+        if not IAnalysisRequest.providedBy(parent) \
+                or IReferenceSample.providedBy(parent):
+            return ""
```

Comment:
Remember there is the `IRequestAnalysis` interface, that is implemented by all analyses that have an analysis request associated (so, reference analyses are excluded, but duplicates are in). It also make this code simpler:
```python
instance = api.get_object(item['obj'])
if not IRequestAnalysis(instance):
    return ""
ar = instance.getRequest()
cooked = ar.getRemarks().replace('\n', '<br/>')
return "<div>{}</div>".format(cooked)
```

Reply:
Yes, but note that I do `instance.getRequest()' instead of `instance.aq_parent`. So even if is a `DuplicateAnalysis`, we get the remarks of the Analysis Request the `DuplicateAnalysis` is associated to.

Reply:
OK, done.